src/sentry/api/endpoints/sentry_app_stats.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.app import tsdb
+
+from sentry.api.base import StatsMixin
+from sentry.api.bases import SentryAppBaseEndpoint, SentryAppStatsPermission
+
+from sentry.models import SentryAppInstallation
+
+
+class SentryAppStatsEndpoint(SentryAppBaseEndpoint, StatsMixin):
+    permission_classes = (SentryAppStatsPermission,)
+
+    def get(self, request, sentry_app):
+        """
+        :qparam float since
+        :qparam float until
+        :qparam resolution - optional
+        """
+
+        query_args = self._parse_args(request)
+
+        installations = SentryAppInstallation.with_deleted.filter(
+            sentry_app=sentry_app, date_added__range=(query_args["start"], query_args["end"])
+        ).values_list("date_added", "date_deleted", "organization_id")
+
+        rollup, series = tsdb.get_optimal_rollup_series(query_args["start"], query_args["end"])
+
+        install_counter = 0
+        uninstall_counter = 0
+
+        install_stats = dict.fromkeys(series, 0)
+        uninstall_stats = dict.fromkeys(series, 0)
+
+        for date_added, date_deleted, organization_id in installations:
+            install_counter += 1
```

Comment:
would this require loading the entire table contents into memory though? or does this not matter since i'm iterating through it once anyway? 

for the uninstalls I would also have to filter the installations again by `date_deleted is not None` to avoid iterating a variable

Reply:
Don't do `len(installations)`. This incurs a separate implicit query with a `COUNT(*)` in it to get the length.

If you wanna do this, you can do like:

```python
installations = list(SentryAppInstallation.....values_list(...))
```

Which will force it to be evaluated, then you can do `len()` without the penalty of another query.

It loads all the rows into memory, but it does it anyways with what you have here. Once you begin iterating, it prefetches and holds it all anyways. There's a way to do it if you were working with really large datasets, but in this case, it's not worth it. In Django you can do an `iterator()` on it, but this still has to load the raw data in memory since it doesn't use a database level cursor to emit rows as iterated.